cmd/powcfg/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	badger "github.com/ipfs/go-ds-badger2"
+	logger "github.com/ipfs/go-log/v2"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	mongods "github.com/textileio/go-ds-mongo"
+	"github.com/textileio/powergate/buildinfo"
+)
+
+var (
+	log    = logger.Logger("powcfg")
+	config = viper.New()
+)
+
+func main() {
+	logger.SetAllLoggers(logger.LevelInfo)
+	log.Infof("starting powcfg:\n%s", buildinfo.Summary())
+
+	if err := wireFlagsAndEnvs(); err != nil {
+		log.Fatalf("wiring flags/envs: %s", err)
+	}
+
+	mongoURI := config.GetString("mongouri")
+	mongoDB := config.GetString("mongodb")
+	mongoCollection := config.GetString("mongocollection")
+	badgerrepo := config.GetString("badgerrepo")
+	dryrun := config.GetBool("dryrun")
```

Comment:
You can supply `--dryrun`, and that would mean that transformation would never write changes.
This is useful to know how many storage configs are going to be affected since this number gets printed when you run the program.